holidays/countries/tuvalu.py

```diff
@@ -0,0 +1,201 @@
+      - `Public Holidays (Amendment) Act 2020 <https://tuvalu-legislation.tv/cms/images/LEGISLATION/AMENDING/2020/2020-0013/2020-0013.pdf>`_
+      - `Public Holidays Act 2 <https://tuvalu-legislation.tv/cms/images/LEGISLATION/PRINCIPAL/1937/1937-0005/1937-0005_2.pdf>`_
+      - `Codes for the representation of names of countries and their subdivisions <https://www.iso.org/obp/ui/#iso:code:3166:TV>`_
+      - `TUVALU-NEWS.TV <https://web.archive.org/web/20140915180104/http://www.tuvalu-news.tv/archives/2007/01/island_special_public_holidays.html>`_
+    """
+
+    country = "TV"
+    default_language = "tvl_TV"
+    # %s (estimated).
+    estimated_label = tr("%s (fakaaoga)")
+    # %s (observed).
+    observed_label = tr("%s (fakamatakuga)")
+    # %s (observed, estimated)
+    observed_estimated_label = "%s (fakamatakuga, fakaaoga)"
+    supported_languages = ("en_GB", "en_US", "tvl_TV")
+    subdivisions = (
+        "FUN",  # Funafuti.
+        "NMG",  # Nanumaga.
+        "NMA",  # Nanumea.
+        "NIT",  # Niutao.
+        "NUI",  # Nui.
+        "NKF",  # Nukufetau.
+        "NKL",  # Nukulaelae.
+        "VAI",  # Vaitupu.
+    )
+    subdivisions_aliases = {
+        # Town/Island Councils.
+        "Funafuti": "FUN",
+        "Nanumaga": "NMG",
+        "Nanumea": "NMA",
+        "Niutao": "NIT",
+        "Nui": "NUI",
+        "Nukufetau": "NKF",
+        "Nukulaelae": "NKL",
+        "Vaitupu": "VAI",
+    }
+    # Tuvalu became fully independent of the United Kingdom on October 1, 1978
+    # Tuvalu's PUBLIC HOLIDAYS (AMENDMENT) ACT 1990 (Act 2 of 1990)
+    # It was first proclaimed on FEB 7th, 1990
+    start_year = 1990
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        kwargs.setdefault("observed_rule", SAT_SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # Public Holidays
+
+        # New Year's Day.
+        self._add_new_years_day(tr("Tausaga Fou"))
+
+        # Commonwealth Day.
+        self._add_holiday_2nd_mon_of_mar(tr("Aso Atefenua"))
+
+        # Gospel Day.
+        self._add_holiday_2nd_mon_of_may(tr("Te Aso o te Tala Lei"))
+
+        # King's Birthday.
+        self._add_holiday_2nd_sat_of_jun(tr("Asofanau Tupu"))
+
+        # National Children's Day.
+        name = tr("Aso Tamaliki")
+
+        if self._year >= 2019:
+            # National Children's Day.
+            self._add_holiday_1_day_past_2nd_sun_of_oct(name)
+
+            # National Youth Day.
+            self._add_holiday_1st_mon_of_aug(tr("Aso tupulaga"))
+        else:
+            # National Children's Day.
+            self._add_holiday_1st_mon_of_aug(name)
+
+        # Tuvalu Day.
+        self._add_holiday_oct_1(tr("Tutokotasi"))
+
+        # Tuvalu Day.
+        self._add_holiday_oct_2(tr("Tutokotasi"))
```

Comment:
```suggestion
        # Tuvalu Day.
        name = tr("Tutokotasi")
        self._add_observed(self._add_holiday_oct_1(name), rule=SAT_SUN_TO_NEXT_MON_TUE)
        self._add_observed(self._add_holiday_oct_2(name), rule=SAT_SUN_TO_NEXT_MON_TUE)
```
Public Holidays Act:
> 3) When either the 2nd day of October or the 26th day of December falls on a Sunday the Monday and Tuesday next following shall be public holidays, and when either day falls on a Monday the Tuesday next following shall be a public holiday.